GliaWidgets/SecureConversations/ChatTranscript/SecureConversations.TranscriptModel+GVA.swift

```diff
@@ -0,0 +1,126 @@
+        // If option contains `url`, then it's URL Button,
+        // otherwise it's Postback Button and option should be sent
+        // to the server as `SingleChoiceOption`
+        if option.destinationPdBroadcastEvent != nil {
+            return broadcastEventButtonAction()
+        } else if let urlString = option.url,
+                  let url = URL(string: urlString) {
+            return urlButtonAction(url: url, urlTarget: option.urlTarget)
+        } else {
+            return postbackButtonAction(for: option)
+        }
+    }
+
+    func postbackButtonAction(for option: GvaOption) -> Cmd {
+        .init { [weak self] in
+            self?.sendGvaOption(option)
+        }
+    }
+}
+
+private extension SecureConversations.TranscriptModel {
+    func urlButtonAction(url: URL, urlTarget: String?) -> Cmd {
+        .init { [weak self] in
+            guard let self else { return }
+
+            let openUrl = { [weak self] url in
+                guard let self = self else { return }
+                guard self.environment.uiApplication.canOpenURL(url) else { return }
+                self.environment.uiApplication.open(url)
+            }
+
+            switch url.scheme?.lowercased() {
+            case URLScheme.tel.rawValue,
+                URLScheme.mailto.rawValue,
+                URLScheme.http.rawValue,
+                URLScheme.https.rawValue:
+                // "tel" ,"mailto" and "http(s)"-based links should be opened by UIApplication
+                openUrl(url)
+
+            default:
+                if urlTarget == .gvaOptionUrlTargetModal ||
+                    urlTarget == .gvaOptionUrlTargetSelf {
+                    // if GvaOption.urlTarget is "modal" or "self", then button url is deeplink
+                    // and should be opened by UIApplication, to provide integrator
+                    // an ability to handle deeplinks they configured.
+                    openUrl(url)
+                } else {
+                    return
+                }
+            }
+        }
+    }
+
+    func broadcastEventButtonAction() -> Cmd {
+        .init { [weak self] in
+            guard let self else { return }
+            self.showAlert(
+                with: self.alertConfiguration.unsupportedGvaBroadcastError,
+                dismissed: nil
+            )
+        }
+    }
+
+    func sendGvaOption(_ option: GvaOption) {
+        let attachment = CoreSdkClient.Attachment(
+            type: .singleChoiceResponse,
+            selectedOption: option.value,
+            options: nil,
+            files: nil,
+            imageUrl: nil
+        )
+
+        let outgoingMessage = OutgoingMessage(
+            content: option.text,
+            attachment: attachment
+        )
+
+        appendItem(
+            .init(kind: .outgoingMessage(outgoingMessage)),
+            to: pendingSection,
+            animated: true
+        )
+
+        _ = environment.sendSecureMessage(
```

Comment:
should we support `cancellable` for sending option during secure conversation?

Reply:
We don't handle `cancellable` at all for sending messages in SecureConversations. So I called it the same way.